disputer/index.js

```diff
@@ -24,73 +24,76 @@ const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
-    message: "Disputer started🔎",
-    empAddress: address,
-    pollingDelay: pollingDelay,
-    priceFeedConfig
-  });
-
-  // Setup web3 accounts an contract instance
-  const accounts = await web3.eth.getAccounts();
-  const emp = await ExpiringMultiParty.at(address);
-
-  // Setup price feed.
-  // TODO: consider making getTime async and using contract time.
-  const getTime = () => Math.round(new Date().getTime() / 1000);
-  const priceFeed = await createPriceFeed(web3, Logger, new Networker(Logger), getTime, priceFeedConfig);
-
-  if (!priceFeed) {
-    throw "Price feed config is invalid";
-  }
+  try {
+    Logger.info({
+      at: "Disputer#index",
+      message: "Disputer started🔎",
+      empAddress: address,
+      pollingDelay: pollingDelay,
+      priceFeedConfig
+    });
+
+    // Setup web3 accounts an contract instance
+    const accounts = await web3.eth.getAccounts();
+    const emp = await ExpiringMultiParty.at(address);
+
+    // Setup price feed.
+    // TODO: consider making getTime async and using contract time.
+    const getTime = () => Math.round(new Date().getTime() / 1000);
+    const priceFeed = await createPriceFeed(web3, Logger, new Networker(Logger), getTime, priceFeedConfig);
+
+    if (!priceFeed) {
+      throw "Price feed config is invalid";
+    }
 
-  // Client and dispute bot
-  const empClient = new ExpiringMultiPartyClient(Logger, ExpiringMultiParty.abi, web3, emp.address);
-  const gasEstimator = new GasEstimator(Logger);
-  const disputer = new Disputer(Logger, empClient, gasEstimator, priceFeed, accounts[0]);
+    // Client and dispute bot
+    const empClient = new ExpiringMultiPartyClient(Logger, ExpiringMultiParty.abi, web3, emp.address);
+    const gasEstimator = new GasEstimator(Logger);
+    const disputer = new Disputer(Logger, empClient, gasEstimator, priceFeed, accounts[0]);
 
-  while (true) {
-    try {
+    while (true) {
       await disputer.queryAndDispute();
       await disputer.queryAndWithdrawRewards();
-    } catch (error) {
-      Logger.error({
-        at: "Disputer#index🚨",
-        message: "Disputer error",
-        error: error
-      });
-    }
-    await delay(Number(pollingDelay));
 
-    if (!shouldPoll) {
-      break;
+      await delay(Number(pollingDelay));
+
+      if (!shouldPoll) {
+        break;
+      }
     }
+  } catch (error) {
+    Logger.error({
+      at: "Disputer#index🚨",
+      message: "Disputer error",
+      error: error.toString()
+    });
+    await delay(5000); // Hacky fix to ensure that winston still fires messages upstream.
   }
 }
 
 const Poll = async function(callback) {
   try {
     if (!process.env.EMP_ADDRESS) {
-      throw new Error(
-        "Bad input arg! Specify an `EMP_ADDRESS ` for the location of the expiring Multi Party within your environment variables."
-      );
+      throw "Bad input arg! Specify an `EMP_ADDRESS ` for the location of the expiring Multi Party within your environment variables.";
```

Comment:
I decided to change the syntax from `throw new Error` to just `throw` as we are catching the error below anyways and the lighter syntax used in this PR formats better.